### PR TITLE
Feature: Package Repository

### PR DESCRIPTION
**Pull request by thetechnick**
### Summary
Introduction of package repositories.

### Change Type
New Feature

### Check List Before Merging

- [ ] This PR passes all pre-commit hook validations.
- [ ] This PR is fully tested and regression tests are included.
- [ ] Relevant documentation has been updated.

### CODE REVIEW

**Comment by kostola** [comment]
Tests implemented in #1289

File: docs/api-reference.md [hunk 1]
```diff
@@ -1560,6 +1701,38 @@ Used in:
...
+
+
+### PackageManifestRepository
+
+PackageManifestRepository contains information about one package repository
+which could be loaded either from a local file or from a container image.
+
+| Field | Description |
+| ----- | ----------- |
+| `file` <br>string | References a file in the filesystem to load. |
```
Comment (thetechnick) [inline docs/api-reference.md:1728]: Does this reference a file or a folder with the repository date inside?
Seems like `folder`

Comment (eqrx) [reply docs/api-reference.md:1728]: Nope, the code right now loads a repository manifest from there by doing `open` on it

**Review by kostola — APPROVED** [review]
I think this is good enough to be merged now :+1:

**Pull request merged**